src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3223,21 +3223,31 @@ def add_vnet_integration(cmd, name, resource_group_name, vnet, subnet, slot=None
 
     list_all_vnets = vnet_client.virtual_networks.list_all()
 
-    vnet_id = ''
+    vnets = []
     for v in list_all_vnets:
-        if v.name == vnet:
-            vnet_id = v.id
-
-    # parsing the arm uri in order to extract vnet_name and vnet_resource_group
-    vnet_id_strings = vnet_id.split('/')
-
-    vnet_resource_group = ''
-    i = 0
-    for z in vnet_id_strings:
-        if z.lower() == "resourcegroups":
-            vnet_resource_group = vnet_id_strings[i + 1]
-        i = i + 1
-
+        if vnet in (v.name, v.id):
+            # parsing the arm uri in order to extract vnet_name and vnet_resource_group
+            vnet_id_strings = v.id.split('/')
```

Comment:
you can use parse_resource_id common utility, example https://github.com/Azure/azure-cli/blob/8adf97355a605d147afc26f6c1b6f6afb6651c2f/src/azure-cli/azure/cli/command_modules/hdinsight/util.py#L14

Reply:
Thanks!